recipes/libgpg-error/libgpg-error.inc

```diff
@@ -5,7 +5,28 @@ COMPATIBLE_HOST_ARCHS = ".*linux"
 
 SRC_URI = "ftp://ftp.gnupg.org/gcrypt/libgpg-error/libgpg-error-${PV}.tar.bz2"
 
-inherit autotools-autoreconf binconfig pkgconfig gettext library
+inherit autotools binconfig pkgconfig gettext library
+
+DEPENDS += "libpthread"
+
+do_compile[prefuncs] += "do_compile_prepend"
+do_compile_prepend() {
+	TARGET_FILE=linux-gnu
+	if [ ${TARGET_OS} != "linux" ]; then
+		TARGET_FILE=${TARGET_OS}
+	fi
+
+	case ${TARGET_ARCH} in
+	  arm-*)	      TUPLE=arm-unknown-linux-gnueabi ;;
+	  i586-*|i686-*)  TUPLE=i686-pc-linux-gnu ;;
+	  x86_64-*)     TUPLE=x86_64-pc-linux-gnu ;;
+	  powerpc-*)	TUPLE=powerpc-unknown-linux-gnu ;;
+	  *)          TUPLE=${TARGET_ARCH}-unknown-linux-gnu ;;
+	esac
+
+	cp ${SRCDIR}/${PN}-${PV}/src/syscfg/lock-obj-pub.$TUPLE.h \
+	  ${SRCDIR}/${PN}-${PV}/src/syscfg/lock-obj-pub.$TARGET_FILE.h
+}
```

Comment:
Eeew, that was not nice.

For now, I think we should accept the above "solution".  The case probably needs to be extended later on.  And possibly needs to be revisisted when upgrading the recipe :disappointed: